utils/swift_build_support/swift_build_support/host_specific_configuration.py

```diff
@@ -0,0 +1,250 @@
+# swift_build_support/host_configuration_support.py -------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+from argparse import ArgumentError
+
+import diagnostics
+
+from .targets import StdlibDeploymentTarget
+
+
+class HostSpecificConfiguration(object):
+
+    """Configuration information for an individual host."""
+
+    def __init__(self, args, host_target):
```

Comment:
Change: Previously `HostSpecificConfiguration` was receiving an `invocation` parameter. Through this parameter, `args` were accessed, and also calculated values like `platforms_to_skip_build`, `platforms_to_skip_test`, `platforms_archs_to_skip_test` and `platforms_to_skip_test_host`. Those calculations were only used by `HostSpecificConfiguration`, but were done by `BuildScriptInvocation`, and only relied on `args.

The solution is passing `args` directly into `HostSpecificConfiguration` and calculate the values from `args` by itself.